src/main/kotlin/org/rust/ide/inspections/import/AutoImportFix.kt

```diff
@@ -87,8 +88,9 @@ class AutoImportFix(element: RsElement) : LocalQuickFixOnPsiElement(element), Hi
 
         fun findApplicableContext(project: Project, path: RsPath): Context<RsPath>? {
             val basePath = path.basePath()
-            if (TyPrimitive.fromPath(basePath) != null) return null
-            if (basePath.reference.multiResolve().isNotEmpty()) return null
+            if (TyPrimitive.fromPath(basePath) != null || basePath.reference.multiResolve().isNotEmpty()) {
+                return findApplicableContextForAssocItemPath(path, project)
```

Comment:
Let's add some comment why we want to import if it can be resolved
It's not obvious that we resolve some incorrect (in compiler sense) paths